jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java

```diff
@@ -16,33 +16,332 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
 
 public class EnvUtilTest {
 
+    @Test
+    public void testConvertTcpToHttpsUrl() {
+        // Given
+        String urlWithHttpsPort = "tcp://0.0.0.0:2376";
+        // When
+        String result1 = EnvUtil.convertTcpToHttpUrl(urlWithHttpsPort);
+        // Then
+        assertEquals("https://0.0.0.0:2376", result1);
+    }
+
+    @Test
+    public void testConvertTcpToHttpUrl() {
+        // Given
+        String urlWithHttpPort="tcp://0.0.0.0:2375";
+        // When
+        String result2 = EnvUtil.convertTcpToHttpUrl(urlWithHttpPort);
+        // Then
+        assertEquals("http://0.0.0.0:2375", result2);
+    }
+
+    @Test
+    public void testExtractLargerVersionWhenBothNull(){
+        assertNull(EnvUtil.extractLargerVersion(null,null));
+    }
+    @Test
+    public void testExtractLargerVersionWhenBIsNull() {
+        //Given
+        String versionA = "4.0.2";
+        //When
+        String result = EnvUtil.extractLargerVersion(versionA,null);
+        //Then
+        assertEquals( versionA, result);
+    }
+    @Test
+    public void testExtractLargerVersionWhenAIsNull() {
+        //Given
+        String versionB = "3.1.1.0";
+        //When
+        String result = EnvUtil.extractLargerVersion(null,versionB);
+        //Then
+        assertEquals( versionB, result);
+    }
+
+    @Test
+    public void testExtractLargerVersion() {
+        //Given
+        //When
+        String result = EnvUtil.extractLargerVersion("4.0.0.1","4.0.0");
+        //Then
+        assertEquals("4.0.0.1",result);
+    }
+
+    @Test
+    public void testGreaterOrEqualsVersionWhenTrue() {
+        //Given
+        String versionA = "4.0.2";
+        String versionB = "3.1.1.0";
+        //When
+        boolean result1 = EnvUtil.greaterOrEqualsVersion(versionA,versionB);
+        //Then
+        assertTrue(result1);
+    }
+
+    @Test
+    public void testGreaterOrEqualsVersionWhenEqual() {
+        //Given
+        String versionA = "4.0.2";
+        //When
+        boolean result2 = EnvUtil.greaterOrEqualsVersion("4.0.2", versionA);
+        //Then
+        assertTrue(result2);
+    }
+
+
+    @Test
+    public void testGreaterOrEqualsVersionWhenFalse() {
+        //Given
+        String versionA = "4.0.2";
+        String versionB = "3.1.1.0";
+        //When
+        boolean result3 = EnvUtil.greaterOrEqualsVersion(versionB,versionA);
+        //Then
+        assertFalse(result3);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    @Ignore
+// TODO: Remove when implementation is fixed
+    public void testGreaterOrEqualsVersionCornerCase() {
```

Comment:
Why are we ignoring this test?

Reply:
![image](https://user-images.githubusercontent.com/488391/85106595-bee12800-b20c-11ea-90ac-5a5fafd08722.png)

Reply:
ah, okay.